backend/Services/DataBuilder.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Internships\Services;
+
+use ECSPrefix20210708\Nette\Neon\Exception;
+use Internships\Interfaces\BuildTool;
+use Internships\Models\PathPair;
+
+abstract class DataBuilder implements BuildTool
+{
+    protected string $workingDirectory;
+    protected static array $paths = [];
```

Comment:
Oh, why would you need static field here?

Reply:
I want to create some manager to guard application from using same path + file combination for different data. I'm currently not sure how to create external class that will keep track of it from multiple builders, but current implementation is not something I want to keep.

And of course, I've just got possible idea how to do it while writing this comment, so I'll better implement it before it fades. External manager and passing paths into constructors should suffice. Path definitions will be moved ouside the Builders, but I think final result will increase readability.